processor/deltatocumulativeprocessor/internal/delta/delta.go

```diff
@@ -0,0 +1,61 @@
+package delta
+
+import (
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/data"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/streams"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+func construct[D data.Point[D]]() streams.Aggregator[D] {
+	acc := &Accumulator[D]{dps: make(map[streams.Ident]D)}
+	lock := &Lock[D]{next: acc}
+	return lock
+}
+
+func Numbers() streams.Aggregator[data.Number] {
+	return construct[data.Number]()
+}
+
+func Histograms() streams.Aggregator[data.Histogram] {
+	return construct[data.Histogram]()
+}
+
+var _ streams.Aggregator[data.Number] = (*Accumulator[data.Number])(nil)
+
+type Accumulator[D data.Point[D]] struct {
+	dps map[streams.Ident]D
+}
+
+func (a *Accumulator[D]) Aggregate(id streams.Ident, dp D) (D, error) {
+	reset := func() {
+		a.dps[id] = dp.Clone()
+	}
+
+	aggr, ok := a.dps[id]
+	if !ok {
+		fmt.Printf("d2c: new stream %s", id)
```

Comment:
Assuming this is just while drafting.

Reply:
removed for now, will add proper logging later